extensions/amp-access/0.1/amp-access.js

```diff
@@ -277,14 +280,24 @@ export class AccessService {
 
   /**
    * @param {string} url
+   * @param {boolean} useAuthData Allows `AUTH(field)` URL var substitutions.
    * @return {!Promise<string>}
    * @private
    */
-  buildUrl_(url) {
+  buildUrl_(url, useAuthData) {
     return this.getReaderId_().then(readerId => {
-      return this.urlReplacements_.expand(url, {
+      const vars = {
         'READER_ID': readerId
-      });
+      };
+      if (useAuthData) {
+        vars['AUTH'] = field => {
+          if (this.authResponse_) {
+            return this.authResponse_[field];
```

Comment:
I replaced the other one with `undefined` - it fits the meaning of what happens better. However, both `null` and `undefined` are explicitly documented to yield empty string in var substitution.

That being said, that triggered me to check UrlReplacements code for this resolution and there was some weirdness there. I fixed it and added more tests in this PR.